facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigRevisionHelper.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.facade.providers;
+
+import org.jboss.pnc.dto.BuildConfigurationRevision;
+import org.jboss.pnc.facade.mapper.api.BuildConfigurationRevisionMapper;
+import org.jboss.pnc.model.BuildConfiguration;
+import org.jboss.pnc.model.BuildConfigurationAudited;
+import org.jboss.pnc.model.GenericEntity;
+import org.jboss.pnc.spi.datastore.repositories.BuildConfigurationAuditedRepository;
+import org.jboss.pnc.spi.datastore.repositories.BuildConfigurationRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.security.PermitAll;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+
+import java.util.Objects;
+
+import javax.transaction.Transactional;
+
+@PermitAll
+@Stateless
+public class BuildConfigRevisionHelper {
+
+    private final Logger logger = LoggerFactory.getLogger(BuildConfigRevisionHelper.class);
+
+    @Inject
+    private BuildConfigurationAuditedRepository buildConfigurationAuditedRepository;
+    @Inject
+    private BuildConfigurationRepository buildConfigurationRepository;
+    @Inject
+    private BuildConfigurationRevisionMapper buildConfigurationRevisionMapper;
+    
+    @Transactional(Transactional.TxType.REQUIRES_NEW)
+    public void updateBuildConfiguration(org.jboss.pnc.model.BuildConfiguration bcEntity) {
+        buildConfigurationRepository.save(bcEntity);
+    }
+
+    @Transactional(Transactional.TxType.REQUIRES_NEW)
+    public BuildConfigurationRevision findRevision(int id, BuildConfiguration bcEntity) {
```

Comment:
Why do we need a new transaction for a lookup?

Reply:
Hmm, what transaction isolation level do we have? Because there is an overlapping transaction in `BuildConfigurationProviderImpl`

Reply:
AFAIK we use the default transaction isolation level, which is Read Commited for Postgres 10: https://www.postgresql.org/docs/10/transaction-iso.html

Reply:
Could you explain more, why we need new transaction here?

As I understand it it is in the same transaction using the same PersistanceContext and therefore should be read from the PersistanceContext itself and won't go to the to read the entity, right?

Reply:
Hmmm, yeah. If we have read committed then we probably don't need the transaction there.

Reply:
Tests passed, seems you are right.